sbin/run_e2e_tests.sh

```diff
@@ -40,50 +40,53 @@ function ctrl_c() {
 
 ##############################
 
-# Check that the all required dotenv files exists.
-reqdotenv "paths" ".paths.env"
-reqdotenv "sidecar" ".sidecar.env"
+WORKSPACE_DIR=$HOME/.spc/workspaces/active_workspace
 
-WORKSPACE_DIR=./workspace-test
-mkdir -p $WORKSPACE_DIR
-cd $WORKSPACE_DIR
+# Copy config files to workspace.
+spc workspace download --config-path "config/spc?ref=siosw/spc-integration" --name e2e
+spc workspace set e2e
 
-echo "Cleaning $WORKSPACE_DIR"
-$SBIN/clean.sh
-# Copy config files to cwd.
-echo "Copying local_devnet config files to cwd..."
-cp -a $CONFIG_DIR/e2e_test/. .
+PATHS_ENV=$WORKSPACE_DIR/.paths.env
+SIDECAR_ENV=$WORKSPACE_DIR/.sidecar.env
+reqdotenv "paths" $PATHS_ENV
+reqdotenv "sidecar" $SIDECAR_ENV
 
 # Start L1
-yes | $SBIN/generate_secrets.sh -d
-yes | $SBIN/start_l1.sh -d -s &
+$SBIN/generate_secrets.sh -dj
+echo "starting l1"
+$SBIN/start_l1.sh -d &>$WORKSPACE_DIR/l1.out &
+
+echo "waiting for host"
 
 # Parse url into host:port
 L1_HOST_AND_PORT=${L1_ENDPOINT#*://}
+echo $L1_HOST_AND_PORT
 # Wait for services
 $SBIN/wait-for-it.sh -t 60 $L1_HOST_AND_PORT | sed "s/^/[WAIT] /"
 echo "L1 endpoint is available"
-until [ -f ".deployed" ]; do
+
+DEPLOYMENTS_ENV=$WORKSPACE_DIR/.deployments.env
+until [ -f $DEPLOYMENTS_ENV ]; do
   echo "waiting for L1 to be fully deployed..."
   sleep 4
 done
-
-reqdotenv "deployments" ".deployments.env"
+reqdotenv "deployments" $DEPLOYMENTS_ENV
 
 # Start sp-geth
-$SBIN/start_sp_geth.sh -c &>proc.out &
-sleep 1
+$SBIN/start_sp_geth.sh -c &>$WORKSPACE_DIR/sp_geth.out &
+sleep 4
```

Comment:
for some reasons the tests fail with `sleep 1` - i would like to get rid of the hardcoded waiting and rely on wait files. should I try implementing this for this PR or should this go in the backlog?

Reply:
I think this is fine for now